crypto/dsa/dsa_ossl.c

```diff
@@ -191,15 +195,11 @@ static int dsa_sign_setup(DSA *dsa, BN_CTX *ctx_in,
      * We do not want timing information to leak the length of k, so we
      * compute g^k using an equivalent exponent of fixed length. (This
      * is a kludge that we need because the BN_mod_exp_mont() does not
-     * let us specify the desired timing behaviour.)
+     * let us specify the desired timing behaviour).
      */
 
-    if (!BN_add(k, k, dsa->q))
+    if (!BN_set_bit(k, BN_num_bits(dsa->q) + 1))
```

Comment:
Does commentary need rewording? I mean it suggests that chosen k adjustment is equivalent modulo q, while suggested replacement adjustment is not. And I don't quite follow why does it have to be one bit wider? Well, it would take two additions to ensure same +1 bit width, but if you omit modulo q equivalence and just force most significant bit, why does it have to be one bit wider? Well, not that it actually matters, because it takes same amount of iterations in BN_mod_exp_mont_consttime either way...

Reply:
I was playing it safe by setting the bit one higher, this guarantees that any check and modulo reduction occurs in constant time.  Just setting the -1 bit doesn't do this because k could be less than dsa->q.  I'll reword the comment.